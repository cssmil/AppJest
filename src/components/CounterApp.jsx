import PropTypes from 'prop-types'
import { useState } from 'react';

export const CounterApp = ({value, name}) => { 
  
  const [counter, setCounter] = useState(value)
  const handleAdd = () => setCounter( counter + 1 );
  const handleSub = () => setCounter( counter - 1 );
  const handleReseT = () => setCounter( value );
  
  return(
    <> 
      <h2>AppTest Jest {name}</h2>
      <h1 >{ counter }</h1>
      <button onClick={ handleAdd } >
        +1
      </button>
      <button onClick={ handleSub } >
        -1
      </button>
      <button aria-label='btn-reset'  onClick={ handleReseT } >
        reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
}

CounterApp.defaultProps = {
  value: 1,
  name: "name test default"
}