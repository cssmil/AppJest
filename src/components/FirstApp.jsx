import PropTypes from 'prop-types'

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
      <h1 className="mt-2"> { title } </h1>
      <hr />
      <p> { subTitle } </p>
      <p> { name } </p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  // name: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: "no tiene titulo",
  subTitle: "No tiene sub-titulo",
  name: "No tiene name"
}