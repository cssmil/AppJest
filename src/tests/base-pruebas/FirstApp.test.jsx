import { describe, test, expect } from '@jest/globals';

import { render } from "@testing-library/react"
import { FirstApp } from '../../components/FirstApp';


describe('Puebas en <FirstApp />', () => { 

  test('Debe hacer match con el snapshot', () => {

    const title = 'Hola, Soy Armando';
    const { container } =render( <FirstApp title={ title } /> );

    expect( container ).toMatchSnapshot()

  });

  test('Debe mostrar el título en un h1', () => {

    const title = 'Hola, Soy Armando';
    const { container } =render( <FirstApp title={ title } /> );

    // const h1 = container.querySelector('h1')
    // expect( h1.innerHTML ).toContain( title )
    
  });

  test('Debe mostrar el sub-título enviado por props', () => {

    const title = 'Hola, Soy Armando';
    const subTitle = 'SOY UN SUB-TITULO';
    const { getByText } =render( 
      <FirstApp 
        title = { title }
        subTitle = { subTitle }
      /> 
    );
    
    expect( getByText(subTitle) ).toBeTruthy();
    
  });

});