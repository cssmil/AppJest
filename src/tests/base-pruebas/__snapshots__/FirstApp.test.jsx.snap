// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    class="mt-2"
  >
     
    Hola, Soy Armando
     
  </h1>
  <hr />
  <p>
     
    No tiene sub-titulo
     
  </p>
  <p>
     
    No tiene name
     
  </p>
</div>
`;
